python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -184,13 +214,20 @@ def evaluate(
         df_grouped_stats = pd.DataFrame()
         # defining variables to compute running mean and variance for t-test computations
         agg_count, agg_mean, agg_M2 = 0, 0, 0
+        group_metric_running_variance_params = {}
         for predictions_dict in test_dataset.map(_predict_fn).take(-1):
             predictions_df = pd.DataFrame(predictions_dict)
 
             # Accumulating statistics for t-test calculation using 1/rank
             clicked_records = predictions_df[predictions_df[self.feature_config.get_label("node_name")] == 1.0]
-            diff = (1/clicked_records[RankingConstants.NEW_RANK] - 1/clicked_records[
-                self.feature_config.get_rank("node_name")]).to_list()
+            clicked_records[RankingConstants.NEW_MRR] = 1.0 / clicked_records[RankingConstants.NEW_RANK]
+            clicked_records[RankingConstants.OLD_MRR] = 1.0 / clicked_records[self.feature_config.get_rank("node_name")]
+            # Statistical analysis pre-processing
+            if len(group_keys) > 0 and len(eval_dict[RankingConstants.METRICS]) > 0:
+                statistical_analysis_preprocessing(clicked_records, group_metric_running_variance_params, group_keys,
```

Comment:
[Minor] In fact, you can create a new object for `BatchedMetrics` which contains different metrics you need for both micro and macro and pass it here instead of a dictionary.

Reply:
I can return it updated.